packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -565,7 +569,8 @@ _flutter.buildConfig = ${jsonEncode(buildConfig)};
     // Insert a random hash into the requests for service_worker.js. This is not a content hash,
     // because it would need to be the hash for the entire bundle and not just the resource
     // in question.
-    final String serviceWorkerVersion = Random().nextInt(4294967296).toString();
+    final String? serviceWorkerVersion =
+        includeServiceWorkerSettings ? Random().nextInt(1 << 32).toString() : null;
```

Comment:
I believe 1 << 32 in JavaScript will not give the expected 4294967296 result (it computes to 1 AFAIK). Looks more readable but not quite as good. Maybe use 1 << 31, or revert to the actual value?